src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,507 @@
+  }
+
+  frameBuffer_.clearDepth(1.0f).clearColor(0,                // color attachment
+                                           Mn::Vector4ui{0}  // clear color
+  );
+
+  CORRADE_INTERNAL_ASSERT(
+      frameBuffer_.checkStatus(Mn::GL::FramebufferTarget::Draw) ==
+      Mn::GL::Framebuffer::Status::Complete);
+}
+
+void CubeMap::mapForDraw(unsigned int index) {
+  frameBuffer_.mapForDraw({
+      {Mn::Shaders::Generic3D::ColorOutput,
+       Mn::GL::Framebuffer::ColorAttachment{index}},
+      // TODO:
+      //{Mn::Shaders::Generic3D::ObjectIdOutput, objectIdAttachment}
+  });
+}
+
+Mn::GL::CubeMapTexture& CubeMap::getTexture(TextureType type) {
+  textureTypeSanityCheck(flags_, type, "CubeMap::getTexture():");
+  return *textures_[type];
+}
+
+#ifndef MAGNUM_TARGET_WEBGL
+// because Mn::Image2D image = textures_[type]->image(...)
+// requires desktop OpenGL
+bool CubeMap::saveTexture(TextureType type,
+                          const std::string& imageFilePrefix) {
+  textureTypeSanityCheck(flags_, type, "CubeMap::saveTexture():");
+
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImageConverter> manager;
+  Cr::Containers::Pointer<Mn::Trade::AbstractImageConverter> converter;
+  if (!(converter = manager.loadAndInstantiate("AnyImageConverter"))) {
+    return false;
+  }
+
+  const char* coordStrings[6] = {"+X", "-X", "+Y", "-Y", "+Z", "-Z"};
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    std::string filename = "";
+    switch (type) {
+      case TextureType::Color: {
+        Mn::Image2D image =
+            textures_[type]->image(convertFaceIndexToCubeMapCoordinate(iFace),
+                                   0, {getPixelFormat(type)});
+        filename = Cr::Utility::formatString("{}.{}.{}.png", imageFilePrefix,
+                                             getTextureTypeFilenameString(type),
+                                             coordStrings[iFace]);
+        if (!converter->exportToFile(image, filename)) {
+          return false;
+        }
+      } break;
+
+      case TextureType::Depth: {
+        filename = Cr::Utility::formatString("{}.{}.{}.hdr", imageFilePrefix,
+                                             getTextureTypeFilenameString(type),
+                                             coordStrings[iFace]);
+        Mn::Image2D image = textures_[type]->image(
+            convertFaceIndexToCubeMapCoordinate(iFace), 0,
+            {Mn::GL::PixelFormat::DepthComponent, Mn::GL::PixelType::Float});
+        Mn::ImageView2D depthAsRedChannel{
+            image.storage(), Mn::PixelFormat::R32F, image.size(), image.data()};
+        if (!converter->exportToFile(depthAsRedChannel, filename)) {
+          return false;
+        }
+        // XXX
+        // display 8 floats
+        /*
+        Mn::Debug{} << "========"
+                    << Cr::Containers::arrayCast<float>(
+                           image.data().prefix(32));
+        Mn::Debug{} << "image size in bytes: " << image.data().size();
```

Comment:
I printed out the size, which was 5760000. This is correct since the image resolution is 1200 x 1200.
 5760000 bytes = 1200 x 1200 x 4bytes/float